packages/rum-core/src/domain/contexts/pageStateHistory.ts

```diff
@@ -77,23 +82,28 @@ export function startPageStateHistory(): PageStateHistory {
   )
 
   return {
-    findAll: (startTime: RelativeTime, duration?: Duration): PageStateServerEntry[] | undefined => {
-      const pageStateEntries = pageStateHistory
-        .findAll(startTime, duration)
-        .reverse()
-        .map((pageState) => {
-          const correctedStartTime = startTime > pageState.startTime ? startTime : pageState.startTime
-          const recenteredStartTime = elapsed(startTime, correctedStartTime)
-
-          return {
-            state: pageState.state,
-            start: toServerDuration(recenteredStartTime),
-          }
-        })
+    findAll: (startTime: RelativeTime, duration: Duration): PageStateServerEntry[] | undefined => {
+      const pageStateEntries = pageStateHistory.findAll(startTime, duration)
+
+      if (pageStateEntries.length === 0) {
+        return
+      }
 
-      if (pageStateEntries.length > 0) {
-        return pageStateEntries
+      const pageStateServerEntries = []
+      const limit = Math.max(0, pageStateEntries.length - maxPageStateEntriesSelectable)
+
+      for (let index = pageStateEntries.length - 1; index >= limit; index--) {
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏It could be easier to grasp as:

```ts
for (let index = pageStateEntries.length - 1; index >= 0 && pageStateServerEntries.length < maxPageStateEntriesSelectable; index--) {
}
```

Or, maybe you could adapt the previous version as: `.findAll().slice(0, maxPageStateEntriesSelectable).reverse().map()`

Reply:
I had the same thought, maybe adding a couple comments could make this code easier to grasp while keeping the most performant approach.

Reply:
I've added some comments: https://github.com/DataDog/browser-sdk/blob/6f8127f31789f1625feda6d2b9fa20ed42e39119/packages/rum-core/src/domain/contexts/pageStateHistory.ts#L84-L88